### ISSUE TITLE
Many features do not work properly. Am I the only one?

### ISSUE THREAD

**Issue by godfuture** [nextcloud/android#4108]
### Actual behaviour
- Loading thumbnails is terribly slow. Way too slow, even being in same network and connected by WiFi.
- Opening favorites section seems to load and stop many times. It looks like it fails, but after many moments of stop and go I see all favorites.
- Streaming mp4 h264 video (maybe others as well) just does not work. It starts to load, but doesn't start to play. If you move the bar, it plays, but fails to fill up additional buffer.
- Opening photos section doesn't work. And once opened, other views will not work anymore.
- Its ironic, but downloaded files takes longer to open. The loading animation while opening spinns for no reason. Weird.
- Uploading many files updates my Android notification area like crazy by opening and closing a notification for every single file, no matter the size. Really creepy.

### Expected behaviour
- thumbs should load fast
- favorites section should load at once
- streaming video should work
- photos section should work and not have influence on other views
- downloaded files should open (nearly) immediately
- only one notification for updates from app
 
### Steps to reproduce
Should be clear from description above.

### Environment data
Android version: 7.1.2

Device model: BQ X5P

Stock or customized system: LineageOS

Nextcloud app version: 3.6.2

Nextcloud server version: 15.0.2

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by szaimen** [comment]
I can aggree with most of your observations and want to add one:
Not only uploading many files produces one notification for each file, but also synchronizing/downloading a folder opens and closes one notification for every file.
Instead it should produce just one notification for the whole folder.

**Comment by albgus** [comment]
No, it's not just you. I find this app very buggy with a lot of silent errors, missing files and upload issues.

I'd rate it 1/5 if Play store would allow me.

**Comment by AndrewwHummer** [comment]
Do you have preview generation enabled on you Nextcloud Server? 
For displaying a large photo collection this can help tremendously, esp. for mobile clients.


_Disclaimer: I am on the 3.7 RC2 Beta, Android 9, Pixel 3XL._
Well, no you are not the only one.
Trying to play a video, instantly crashes my App
Opening the Fotos section, it loads a long time then nothing appears. Sometimes it crashes.

But I expect the app to open a notification for every file it uploads no matter the size. An option to disable that would be nice.

But as for the core functionality, (auto) uploading and handling files, the app works more or less flawlessly in my experience.

**Comment by hkingman** [comment]
Nextcloud noob here, weighing in "FWIW" (heh). 

tldr; The upload process is fraught with routine, 15-minute delays, unless you disable upload notifications and enable high-performance mode. The app needs an "initial sync" wizard to steer noobs like me around pitfalls such as this (I fell into quite a few others! read on if interested...)

tldr2; When uploading files, the nextcloud app appears to first make a complete copy *on the phone itself*. This beggars my expectations of a cloud "sync" platform. Why couldn't it just attempt to pull the original, falling back on download/local cache when needed, like the old "redirect 404s to PHP" caching trick Rasmus invented circa 1998, heh.

tldr3; The Android nextcloud app has issues with the Oreo window manager on my device. If I switch to another app, it does not appear as a card in the stack of running apps. I have to open it from the desktop icon every time. 

I'm self-hosting nextcloud on BSD (FreeNAS), and it was an easy install thanks to [Samuel Dowling's awesome howto](https://www.samueldowling.com/2018/12/08/install-nextcloud-on-freenas-iocage-jail-with-hardened-security/). [Nextcloud's scan tool](https://scan.nextcloud.com/) give the installation an "A". 

Problems arose when trying the Android app:

1. I went to AutoUploads, and set up various phone folders to "sync" to nextcloud -- nothing happened. I net-searched and found that AutoUploads only handles new items, and that I must "upload" old content. So, the words "sync" and "AutoUpload" are misleading. "Copy new items going forward" would be more accurate. 

Seemingly every new nextcloud user goes through this; it's almost like a hazing ritual. There appear to be many years-old threads claiming this will be fixed "any day now." 

I can actually see a case for handling things this way: a true rsync-like "sync" function would tend to "un-clobber" server-side deletions and clobber edits. 

I suppose what's really needed is some type of first-use wizard that handles the initial sync -- and this should definitely *not* try to just re-use the quirky existing upload function. It should use rsync or similar -- read on for my reasoning.

2. I next used the app to visit my nextcloud folder, selected upload, navigated to my photos folder, hit the `&vellip;` menu and "select all" 2,500 photos and videos. An eon seemed to pass, with nothing appearing to happen. I reloaded the nextcloud folder on my phone and on a PC several times to confirm that nothing is happening. I searched the server logs for errors, nothing. Finally, a notification about an upload in progress popped up. I clicked on it, and was taken to the Android app's "Uploads" page. Now, things start to chug. Four pics upload. Then nothing. If I reload the page, four more go. This is going to take forever. 

I go into phone settings, search "build number" and click on it seven times to enable the developer menu. I go into it, and set the phone not to sleep while plugged in. I leave it open to the "Uploads" page overnight. About 1,000 files are transferred, albeit with routine delays of 15 minutes. And then.... the phone's storage becomes full, and I awaken to alarming messages and have to delete a few installed apps to regain the use of my device. 

3. Upon discovering that the nextcloud app actually duplicates content before uploading it, I realize I need to have it store stuff on my device's encrypted SD card. I make that change in the settings, and the app freezes on the pin entry screen. I wait for 15 minutes before closing it and relaunching. At that point, I discover that the 1,500 or so pending uploads failed because they are "not found" -- the app wasn't smart enough to change the path when I moved the storage. This is frustrating, because in order to get all of my content uploaded, I know of no other option but to manually click 1,000 checkboxes, or "select all" again and let the device re-upload the first 1,000 files that it already uploaded (I chose this option). 

Again, I wish the initial upload process benefited from something like rclone (rsync over webdav) that was smart enough not to re-upload the same files all over again. While the new, redundant upload progressed in fits and starts, I happened upon this thread, and it occurred to me to try disabling notifications. I opened my drop-down notifications menu, held my finger on the nextcloud upload notification, and then selected "don't show this kind of notification" or whatever. This resulted in about 20-30 files getting uploaded between each 15-minute pause. Baby steps. Next, I tried enabling "high-performance mode", and so far, that appears to have been the ticket. 

The nextcloud platform has so much promise, and the Android app appears to be a great start. But, it'll have to get a bit smoother if everyday idiots like myself ;-) are ever going to see it as a viable G-Suite alternative

**Comment by szaimen** [comment]
This issue is not fixed yet!

**Comment by tobiasKaminsky** [comment]
I think there are issues for all addressed problems.
Having such a meta issue will not help in debugging problems, so please look up the individual issues and reply there.
E.g. for photo view, there is now https://github.com/nextcloud/android/pull/2250 which needs to be tested/feedback is welcome.

If there are issues that are not yet addressed, please create new ones.

(As I am currently the only employee on Nextcloud working on this Android app, things unfortunately do not go as fast as wished…).
